inst/valreports/UnitTestReport.Rmd

```diff
@@ -0,0 +1,69 @@
+---
+title: "Automated Unit Testing Report"
+output:
+  word_document: default
+  html_document: default
+---
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE, message = FALSE, results = "asis")
+library(dplyr)
+
+SummariseResults <- function(res) {
+  res %>% summarise_if(is.numeric, sum) %>% knitr::kable()
+}
+WriteResults <- function(res) {
+  purrr::pwalk(list(res$file, res$Categories), WriteFileResults)
+}
+WriteFileResults <- function(file, categories, ...) {
+  cat("###", file, "\n")
+  purrr::pwalk(list(categories$context, categories$Results), WriteCategoryResults)
+}
+WriteCategoryResults <- function(category, tests, ...) {
+  cat("####", category, "\n")
+  print(knitr::kable(tests))
+}
+path0 <- here::here()
+
+if(!require(KinGUI)) devtools::load_all(path0)
+library(devtools)
+#library(golem)
+```
+
+```{r include=FALSE}
+devtools::test() ## This block is needed due to produce the correct report. Donot remove it. 
+```
+
+
+
+```{r GenerateResults}
+file_lists <- list.files(paste0(path0,"/tests/testthat/"))
+if("_snaps" %in% file_lists) {file_lists <- file_lists[file_lists!="_snaps"]}
+testResultsRaw_list <- lapply(file_lists,function(x)testthat::test_file(paste0(path0,"/tests/testthat/",x), reporter = testthat::ListReporter))
```

Comment:
If we want to handle arbitrary R packages we won't be able to assume that the package uses `testthat` as its testing framework. 

I'm also a _bit_ nervous about re-implementing `testthat::test_package` with our own handlers. `testthat` has some pretty strong assumptions on file names that we might also need to consider here. (`setup-*.R` files, and arbitrary `fixtures` files for test data. I think only `test-*` files actually get tested).

Reply:
It is not intended to do that but to give an example of what can be included in a single package testing protocol and testing implementation. Assuming that tests has been run by the reference image, could we take the same strategy as what has been implemented in the reference image?